src/solc_0.8/avatar/Avatar.sol

```diff
@@ -0,0 +1,78 @@
+//SPDX-License-Identifier: MIT
+// solhint-disable-next-line compiler-version
+pragma solidity 0.8.2;
+
+import {AvatarBase} from "./AvatarBase.sol";
+import {IAvatarMinter} from "../common/interfaces/IAvatarMinter.sol";
+import {Upgradeable} from "../common/BaseWithStorage/Upgradeable.sol";
+import {IMintableERC721} from "../common/interfaces/@maticnetwork/pos-portal/root/RootToken/IMintableERC721.sol";
+import {ERC721Upgradeable} from "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+
+/// @title This contract is a erc 721 compatible NFT token that represents an avatar and can be minted by a minter role.
+/// @dev This contract support meta transactions.
+/// @dev This contract is final, don't inherit form it.
+contract Avatar is AvatarBase, Upgradeable, IMintableERC721 {
```

Comment:
What's your approach to `function setTokenMetadata(uint256 tokenId, bytes memory data) external` as per IRootERC721.sol?

Reply:
I'm not using metadata.